final_assessment/task_1.py

```diff
@@ -0,0 +1,86 @@
+# TASK 1
+#
+# Design a parent class called CFGStudent.
+#
+# It should have general attributes like name, surname, age, email, student id
+# and methods to fetch student’s full name and student’s id.
+#
+# Design a child class called NanoStudent, which  inherits from CFGStudent class.
+# This class should have exactly the same attributes as its parent class,
+# as well as an additional one called ‘specialization’ and course grades.
+#
+# The child class ‘generate_id’ method should override its parent method to add the suffix ‘NANO’
+# in front of the id.
+#
+# New methods ‘add_new_grade’ and ‘get_course_grades’ should be added.
+# You can use  it as a skeleton code for your classes OR adjust it and create your own.
+#
+# SEE NOTES BELOW
```

Comment:
5/15- Partially correct class definitions for CFGStudent and Nanostudent. But fundamental issues with the class definitions mean all test cases fail. Also some misreadings of the problem (e.g. IDs were meant to be a random number between 1000 and 10000 but can see incrementing values here)